### PR TITLE
Draft: new wasm findex

### PR DESCRIPTION
**Pull request by bgrieder**
Calls the new Findex WASM API
Needs http://gitlab.cosmian.com/core/findex/-/merge_requests/35 to be merged first

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "dependencies": {
         "@types/uuid": "^8.3.4",
         "cosmian_cover_crypt": "^6.0.8",
-        "cosmian_findex": "^0.7.2",
+        "cosmian_findex": "file:../findex/pkg/bundler",
```
Comment (ThibaudDauce) [inline package.json:24]: Need to change this after the merge of the MR in Findex

File: src/crypto/sse/findex/simple.ts [hunk 2]
```diff
@@ -245,41 +252,59 @@ export async function upsert(
...
   }
 
+  const kws: Uint8Array[] = []
+  keywords.forEach(k => {
+    if (k instanceof Uint8Array) {
+      kws.push(k)
+    } else {
+      kws.push(new TextEncoder().encode(k))
+    }
+  })
```
Comment (ThibaudDauce) [inline src/crypto/sse/findex/simple.ts:280]:
```suggestion
  const kws = keywords.map((k) => k instanceof Uint8Array ? k : new TextEncoder().encode(k));
```

File: src/crypto/sse/findex/simple.ts [hunk 3]
```diff
@@ -245,41 +252,59 @@ export async function upsert(
...
+    },
+    async (uids: Uint8Array[]) => {
+      return await fetchChains(uids)
     }
   )
 
-  return deserializeList(serializedIndexedValues).map(
-    (bytes) => new IndexedValue(bytes)
-  )
+  return serializedIndexedValues.map(bytes => { return new IndexedValue(bytes) })
```
Comment (ThibaudDauce) [inline src/crypto/sse/findex/simple.ts:309]:
```suggestion
  return serializedIndexedValues.map(bytes => new IndexedValue(bytes))
```

File: src/utils/utils.ts [hunk 4]
```diff
@@ -100,14 +104,37 @@ function getSizeNumberOfBytes(stream: Uint8Array): number {
...
+//       sizeNumberOfBytes,
+//       sizeNumberOfBytes + itemLen
+//     )
+//     serializedItems = serializedItems.slice(sizeNumberOfBytes + itemLen)
+//     items.push(item)
+//   }
+//   return items
+// }
+
+
```
Comment (ThibaudDauce) [inline src/utils/utils.ts:130]: Remove this code

File: package.json (hunk 1 above)
Comment (bgrieder) [reply package.json:24]: yes

**Pull request merged**